src/nccl_ofi_interface_nvidia.c

```diff
@@ -7,6 +7,44 @@
 #include "nccl_ofi.h"
 #include "nccl_ofi_api.h"
 
+static ncclResult_t getProperties_v8(int dev_id, ncclNetProperties_v8_t* props)
+{
+	nccl_ofi_properties_t ofi_properties;
+	ncclResult_t ret = nccl_net_ofi_get_properties(dev_id, &ofi_properties);
+	if (ret != ncclSuccess) {
+		return ret;
+	}
+
+	props->name = ofi_properties.name;
+	props->pciPath = ofi_properties.pci_path;
+	props->guid = ofi_properties.guid;
+	props->ptrSupport = NCCL_PTR_HOST;
+	if (ofi_properties.hmem_support) {
+		props->ptrSupport |= NCCL_PTR_CUDA;
+	}
+	if (ofi_properties.dmabuf_support) {
+		props->ptrSupport |= NCCL_PTR_DMABUF;
+	}
+	/*
+	 * Signal to NCCL that a registered MRs is not tied to a particular
+	 * communicator.
+	 *
+	 * The plugin just set this to `true` because the plugin creates a
+	 * libfabric EP per NIC (rail) and a MR is registered on the EPs of all
+	 * rails (See `reg_mr_ep()` for the RDMA protocol and
+	 * `register_mr_buffers()` for the SENDRECV protocol).
+	 */
+	props->regIsGlobal = 1;
```

Comment:
I think this one is complicated. If NCCL initializes the plugin in parallel, it will create a separate endpoint for each initializing thread (https://github.com/aws/aws-ofi-nccl/blob/213172a71dd65addd329f643ac8dea9078e33937/src/nccl_ofi_rdma.c#L5549-L5551). In this case, communicators may be on different endpoints.

Confusingly, that all doesn't matter for EFA because EFA does not use the `FI_MR_ENDPOINT` mode, so memory registrations are per-domain rather than per-endpoint. But for providers that do set `FI_MR_ENDPOINT`, I think `regIsGlobal` should actually be 0.

Reply:
Correct, we need to look at the value of `FI_MR_ENDPOINT` to set `regIsGlobal`.

Reply:
> If NCCL initializes the plugin communicators in parallel, it will create a separate endpoint for each initializing thread 

From 

https://github.com/NVIDIA/nccl/blob/b6475625fbcaa2c3c0e50eed2fa1255d7514d4a2/src/register.cc#L56-#L80

NCCL creates a loopback communicator just for the sake of creating a NIC-global registration. If a provider did not tie registrations to endpoints, we'd still register user-provided buffers (comm-specific) against the domain that was created during plugin initialization (once per rank via ncclGetUniqueId() which the comm in question is bound to), no?

Agreed on the FI_MR_ENDPOINT check though.